dipy/reconst/sfm.py

```diff
@@ -33,7 +33,76 @@
     warnings.warn(w)
 
 
-def sfm_design_matrix(gtab, sphere, response, mode='signal'):
+# Isotropic signal models: these are models of the part of the signal that
+# changes with b-value, but does not change with direction. This collection is
+# extensible, by inheriting from IsotropicModel/IsotropicFit below:
+class IsotropicModel(ReconstModel):
+    """
+    A base-class for the representation of isotropic signals.
+    The default behavior, suitable for single b-value data is to calculate the
+    mean in each voxel as an estimate of the signal that does not depend on
+    direction.
+    """
+    def __init__(self, gtab):
+        ReconstModel.__init__(self, gtab)
+
+    def fit(self, data):
+        data_no_b0 = data[..., ~self.gtab.b0s_mask]
+        if np.sum(self.gtab.b0s_mask) > 0:
+            s0 = np.mean(data[..., self.gtab.b0s_mask], -1)
+            to_fit = data_no_b0 / s0[..., None]
+        else:
+            to_fit = data_no_b0
+        params = np.mean(to_fit, -1)
+        if len(data_no_b0.shape) == 1:
+            n_vox = 1
+        else:
+            n_vox = data_no_b0.shape[0]
+        return IsotropicFit(self, params, n_vox)
+
+
+class IsotropicFit(ReconstFit):
+    def __init__(self, model, params, n_vox):
+        self.model = model
+        self.params = params
+        self.n_vox = n_vox
+
+    def predict(self, gtab=None):
+        if gtab is None:
+            gtab = self.model.gtab
+        return self.params[..., np.newaxis] + np.zeros((self.n_vox,
+                                                        np.sum(~gtab.b0s_mask)))
+
+class ExponentialIsotropicModel(IsotropicModel):
+    def fit(self, data):
+        if len(data.shape) == 1:
```

Comment:
Again, my preference would be: standardize the data and then write a single algorithm without exceptions